staging/src/k8s.io/cli-runtime/pkg/resource/builder.go

```diff
@@ -252,10 +253,15 @@ func (b *Builder) FilenameParam(enforceNamespace bool, filenameOptions *Filename
 			}
 			b.URL(defaultHttpGetAttempts, url)
 		default:
-			if !recursive {
+			matches, err := expandIfFilePattern(s)
```

Comment:
I'd like to add a `klog.V(4)` with the matches here for future debugging sake but we probably shouldn't log from library code. Does this log at the caller in kubectl?

Reply:
I believe cmds are not logging the source uri/filename. If not here it would have to be done on each cmd since every other reusable place is also in the `cli-runtime` pkg.

Reply:
After some thought let's go ahead and add the log statement please.

Reply:
I don't know what the log statement is for, but if the goal is to see what happens, then why wouldn't we suggest using `--dry-run=client` of some sort?

Reply:
there is no output about the file that picked up the resource so it would go like
```
kubectl --dry-run=client apply -f '*.yaml'
deployment.apps/foo created (dry run)
persistentvolumeclaim/bar created (dry run)
[...]
```
I suppose it makes more sense if the pattern is complex or you have more than 1 then you would know why `foo.yaml` and `bar.yaml` were used.